taskcluster/ac_taskgraph/loader/build_config.py

```diff
@@ -93,24 +99,32 @@ def loader(kind, path, config, params, loaded_tasks):
     # Build everything unless we have an optimization strategy (defined below).
     files_changed = []
     affected_components = ALL_COMPONENTS
+    component_dependencies = defaultdict(set)
+    reverse_component_dependencies = defaultdict(set)
+
+    for component in sorted(get_components()):
+        name = component["name"]
+        for dep_component in get_upstream_deps_for_component(name):
```

Comment:
I was trying to figure out what line could have increased the duration that much, and I found it 😁 

I read the previous proposal about baking this in `toolchain-linux64-android-gradle-dependencies`. I've thought about it out loud to understand the implications of such a change. While I believe the happy path easily works, supporting edge cases will be tough. I'm going to expand on this in the summary of my review. 

In the meantime, I believe there is a lower-hanging fruit. If I'm not mistaken, every time we run gradle, we spend several seconds initializing everything, even though the task we're interested in takes less than a second. My proposal is: instead of running `./gradlew X:dependencies --configuration implementation` a hundred times, we can craft a single command line that generates everything at once. I tried with this example, and it worked fine locally: `./gradlew browser-session:dependencies --configuration implementation browser-menu:dependencies --configuration implementation` (You need to repeat `--configuration implementation` for each component, otherwise you end up with an output that is too verbose).

Pros: 
 * We know we have to process all components, now. So it's easy to craft one big command.
 * We shave off the init steps that last several seconds.

Cons: 
 * The command line may be too large for gradle to comprehend. We may need to split it into 2
 * We have to make a smarter parser since the output will contain the whole tree.

I wouldn't be surprised if we decision task would take less than 7 minutes to run (instead of 10). That said, we have to test this out. I see you already have some baseline results and a testing protocol. I'm happy to defer this testing to you if you have everything set up. Otherwise, I can take care of it. What do you think @mozbhearsum ?

Reply:
To be clear: you're suggesting we run one gradle command for all (or as many as we can) components at once, instead of one at a time? I think that makes sense, and might greatly reduce the pressure for any additional optimization. I'll dig into it more.

Reply:
I tried this, and it basically reduces gradle time to nothing on my laptop -- like, <5s to get the entire tree.